lib/removeUnusedMaterials.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = removeUnusedMaterials;
+
+function removeUnusedMaterials(gltf, stats) {
+    var usedMaterialIds = {};
+    var meshes = gltf.meshes;
+
+    // Build hash of used materials by iterating through meshes
+    if (defined(meshes)) {
+        for (var meshId in meshes) {
+            if (meshes.hasOwnProperty(meshId)) {
+                if (defined(meshes[meshId].primitives)) {
+                    var length = meshes[meshId].primitives.length;
+                    for (var i = 0; i < length; i++) {
+                        if (defined(meshes[meshId].primitives[i].material)) {
```

Comment:
`material` is required so this test isn't needed.

See https://github.com/KhronosGroup/glTF/tree/master/specification#primitivematerial-white_check_mark